File: 1647-minimum-deletions-to-make-character-frequencies-unique/1647-minimum-deletions-to-make-character-frequencies-unique.java
class Solution {
    public int minDeletions(String s) {
        int[] freq = new int[26];
        for(char c : s.toCharArray()) freq[c-'a']++;
        Arrays.sort(freq);
        int ans = 0;
        for(int i=24; i>=0; i--){
            if(freq[i] >= freq[i+1]){
                int tmp = freq[i];
                freq[i] = Math.max(0, freq[i+1]-1);
                ans += (tmp-freq[i]);
            }
        }
        return ans;
    }
}

File: 1647-minimum-deletions-to-make-character-frequencies-unique/1647-minimum-deletions-to-make-character-frequencies-unique.py
class Solution:
    def minDeletions(self, s: str) -> int:
        freq = [0]*(26)
        for c in s:
            freq[ord(c)-ord('a')] += 1
        freq.sort()
        ans = 0
        for i in range(24, -1, -1):
            if freq[i] >= freq[i+1]:
                tmp = freq[i]
                freq[i] = max(0, freq[i+1]-1)
                ans += (tmp-freq[i])
        return ans